### PR TITLE
[Merged by Bors] - feat(RingTheory.Valuation.Minpoly): add valuation lemmas

### PR DESCRIPTION
**Pull request by mariainesdff**
We prove some results about valuations of zero coefficients of minimal polynomials.

Co-authored-by: Filippo A. E. Nuccio <filippo.nuccio@univ-st-etienne.fr>

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Valuation/Basic.lean [hunk 1]
```diff
@@ -211,6 +211,9 @@ theorem ne_zero_iff [Nontrivial Γ₀] (v : Valuation K Γ₀) {x : K} : v x ≠
 theorem unit_map_eq (u : Rˣ) : (Units.map (v : R →* Γ₀) u : Γ₀) = v u :=
   rfl
 
+theorem ne_zero_of_unit [Nontrivial Γ₀] (v : Valuation K Γ₀) (x : Kˣ) : v x ≠ (0 : Γ₀) := by
+  simp only [ne_eq, Valuation.zero_iff, Units.ne_zero x, not_false_iff]
```
Comment (jcommelin) [inline Mathlib/RingTheory/Valuation/Basic.lean:215]: Could you please add a variant that assumes `IsUnit x` for `x : K` (as opposed to a bundled unit)?

File: Mathlib/RingTheory/Valuation/Minpoly.lean [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+open FiniteDimensional minpoly Polynomial
+
+variable {K : Type*} [Field K] {Γ₀ : Type*} [LinearOrderedCommGroupWithZero Γ₀]
+  (v : Valuation K Γ₀) (L : Type*) [Field L] [Algebra K L]
+
+namespace Valuation
+
+/-- For `x ∈ K` the valuation of the zeroth coefficient of the minimal polynomial
+of `algebra_map K L x` over `K` is equal to the valuation of `x`.-/
+theorem coeff_zero_minpoly (x : K) : v ((minpoly K (algebraMap K L x)).coeff 0) = v x := by
```
Comment (jcommelin) [inline Mathlib/RingTheory/Valuation/Minpoly.lean:33]: This can be a simp lemma, right?

Comment (faenuccio) [reply Mathlib/RingTheory/Valuation/Minpoly.lean:33]: Done

File: Mathlib/RingTheory/Valuation/Basic.lean (hunk 1 above)
Comment (faenuccio) [reply Mathlib/RingTheory/Valuation/Basic.lean:215]: Done!

**Review by faenuccio — COMMENTED** [review]
Thanks, I have merged master and addressed your comments.

File: Mathlib/RingTheory/Valuation/Minpoly.lean [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+@[simp]
+theorem coeff_zero_minpoly (x : K) : v ((minpoly K (algebraMap K L x)).coeff 0) = v x := by
+  rw [minpoly.eq_X_sub_C, coeff_sub, coeff_X_zero, coeff_C_zero, zero_sub, Valuation.map_neg]
+
+variable {L}
+
+/- For any unit `x : Lˣ`, we prove that a certain power of the valuation of zeroth coefficient of
+the minimal polynomial of `x` over `K` is nonzero. This lemma is helpful for defining the valuation
+on `L` inducing `v`.-/
+theorem pow_coeff_zero_ne_zero_of_unit [FiniteDimensional K L] (x : Lˣ) :
```
Comment (jcommelin) [inline Mathlib/RingTheory/Valuation/Minpoly.lean:42]: Stating this for `x : L` and `h : IsUnit x` will also allow it to apply slightly more generally in situations where you don't have a bundled unit. Note that the type never refers to `x`, but only to `x.1`.

Comment (faenuccio) [reply Mathlib/RingTheory/Valuation/Minpoly.lean:42]: Done

**Pull request closed without merging**